storage/bootstrap/bootstrapper/commitlog/source_prop_test.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package commitlog
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
+	"github.com/spaolacci/murmur3"
+
+	"github.com/m3db/m3db/context"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/persist/fs/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+	"github.com/m3db/m3db/ts"
+	xtime "github.com/m3db/m3x/time"
+)
+
+const maxShards = 8192
+
+func TestCommitLogSourcePropCorrectlyBootstrapsFromCommitlog(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.Rng.Seed(123456789)
+	parameters.MinSuccessfulTests = 100
+	props := gopter.NewProperties(parameters)
+
+	props.Property("Commitlog bootstrapping properly bootstraps the entire commitlog", prop.ForAll(
+		func(input propTestInput) (bool, error) {
+			// Make sure we have a new directory for each test run
+			dir, err := ioutil.TempDir("", "foo")
+			if err != nil {
+				return false, err
+			}
+			defer func() {
+				os.RemoveAll(dir)
+			}()
+
+			// Configure the commitlog to use the test directory and set the blocksize
+			commitLogOpts := commitlog.NewOptions().
+				SetBlockSize(2 * time.Hour).
+				SetFilesystemOptions(fs.NewOptions().SetFilePathPrefix(dir))
+			bootstrapOpts := testOptions().SetCommitLogOptions(commitLogOpts)
+
+			// Instantiate commitlog
+			log, err := commitlog.NewCommitLog(commitLogOpts)
+			if err != nil {
+				return false, err
+			}
+			err = log.Open()
+			if err != nil {
+				return false, err
+			}
+
+			// Write all the datapoints to the commitlog
+			for _, write := range input.writes {
+				log.Write(context.NewContext(), write.series, write.datapoint, write.unit, write.annotation)
```

Comment:
do you need to check the error returned here?

Reply:
fixed